"zip END header not found" on pushRemoteCache

```scala
[IJ]> pushRemoteCache
java.util.concurrent.ExecutionException: java.util.zip.ZipError: zip END header not found
        at java.util.concurrent.FutureTask.report(FutureTask.java:122)
        at java.util.concurrent.FutureTask.get(FutureTask.java:192)
        at sbt.ConcurrentRestrictions$$anon$4.take(ConcurrentRestrictions.scala:311)
        at sbt.Execute.next$1(Execute.scala:117)
        at sbt.Execute.processAll(Execute.scala:127)
        at sbt.Execute.runKeep(Execute.scala:95)

[error] java.util.concurrent.ExecutionException: java.util.zip.ZipError: zip END header not found




        at sbt.EvaluateTask$.liftedTree1$1(EvaluateTask.scala:478)
        at sbt.EvaluateTask$.run$1(EvaluateTask.scala:477)
        at sbt.EvaluateTask$.runTask(EvaluateTask.scala:500)
        at sbt.internal.Aggregation$.$anonfun$timedRun$4(Aggregation.scala:103)
        at sbt.EvaluateTask$.withStreams(EvaluateTask.scala:390)
        at sbt.internal.Aggregation$.timedRun(Aggregation.scala:101)
        at sbt.internal.Aggregation$.runTasks(Aggregation.scala:115)
        at sbt.internal.Aggregation$.$anonfun$applyTasks$1(Aggregation.scala:70)
        at sbt.Command$.$anonfun$applyEffect$2(Command.scala:145)
        at sbt.internal.Aggregation$.$anonfun$evaluatingParser$11(Aggregation.scala:230)
        at sbt.internal.Act$.$anonfun$actParser0$8(Act.scala:494)
        at sbt.Command$.process(Command.scala:189)
        at sbt.MainLoop$.$anonfun$processCommand$5(MainLoop.scala:245)
        at scala.Option.getOrElse(Option.scala:189)
        at sbt.MainLoop$.process$1(MainLoop.scala:245)
        at sbt.MainLoop$.processCommand(MainLoop.scala:276)
        at sbt.MainLoop$.$anonfun$next$5(MainLoop.scala:163)
        at sbt.State$StateOpsImpl$.runCmd$1(State.scala:289)
        at sbt.State$StateOpsImpl$.process$extension(State.scala:325)
        at sbt.MainLoop$.$anonfun$next$4(MainLoop.scala:163)
        at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
        at sbt.MainLoop$.next(MainLoop.scala:163)
        at sbt.MainLoop$.run(MainLoop.scala:144)
        at sbt.MainLoop$.$anonfun$runWithNewLog$1(MainLoop.scala:119)
        at sbt.io.Using.apply(Using.scala:27)
        at sbt.MainLoop$.runWithNewLog(MainLoop.scala:112)
        at sbt.MainLoop$.runAndClearLast(MainLoop.scala:66)
        at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:51)
        at sbt.MainLoop$.runLogged(MainLoop.scala:42)
        at sbt.StandardMain$.runManaged(Main.scala:192)
        at sbt.xMain$.$anonfun$run$8(Main.scala:101)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
        at scala.Console$.withIn(Console.scala:230)
        at sbt.internal.util.Terminal$.withIn(Terminal.scala:560)
        at sbt.internal.util.Terminal$.$anonfun$withStreams$1(Terminal.scala:350)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
        at scala.Console$.withOut(Console.scala:167)
        at sbt.internal.util.Terminal$.$anonfun$withOut$2(Terminal.scala:550)
        at scala.util.DynamicVariable.withValue(DynamicVariable.scala:62)
        at scala.Console$.withErr(Console.scala:196)
        at sbt.internal.util.Terminal$.withOut(Terminal.scala:550)
        at sbt.internal.util.Terminal$.withStreams(Terminal.scala:350)
        at sbt.xMain$.run(Main.scala:86)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sbt.internal.XMainConfiguration.run(XMainConfiguration.java:56)
        at sbt.xMain.run(Main.scala:46)
        at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
        at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
        at xsbt.boot.Launch$.run(Launch.scala:109)
        at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
        at xsbt.boot.Launch$.launch(Launch.scala:117)
        at xsbt.boot.Launch$.apply(Launch.scala:18)
        at xsbt.boot.Boot$.runImpl(Boot.scala:56)
        at xsbt.boot.Boot$.main(Boot.scala:18)
        at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.util.zip.ZipError: zip END header not found
        at sbt.internal.inc.zip.ZipCentralDir.zerror(ZipCentralDir.java:305)
        at sbt.internal.inc.zip.ZipCentralDir.findEND(ZipCentralDir.java:249)
        at sbt.internal.inc.zip.ZipCentralDir.initCEN(ZipCentralDir.java:134)
        at sbt.internal.inc.zip.ZipCentralDir.<init>(ZipCentralDir.java:86)
        at sbt.internal.inc.IndexBasedZipFsOps$.readCentralDir(IndexBasedZipFsOps.scala:31)
        at sbt.internal.inc.IndexBasedZipFsOps$.readCentralDir(IndexBasedZipFsOps.scala:26)
        at sbt.internal.inc.IndexBasedZipOps.mergeArchives(IndexBasedZipOps.scala:169)
        at sbt.internal.inc.IndexBasedZipOps.mergeArchives(IndexBasedZipOps.scala:86)
        at sbt.internal.inc.IndexBasedZipOps.includeInArchive(IndexBasedZipOps.scala:103)
        at sbt.internal.inc.JarUtils$.includeInJar(JarUtils.scala:160)
        at sbt.internal.RemoteCache$.$anonfun$configCacheSettings$1(RemoteCache.scala:167)
        at scala.Function1.$anonfun$compose$1(Function1.scala:49)
        at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
        at sbt.std.Transform$$anon$4.work(Transform.scala:68)
        at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
        at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
        at sbt.Execute.work(Execute.scala:291)
        at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
        at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

[error] Use 'last' for the full log.
```

--- a/main/src/main/scala/sbt/internal/RemoteCache.scala
+++ b/main/src/main/scala/sbt/internal/RemoteCache.scala
@@ -163,7 +163,8 @@ object RemoteCache {
           val artp = artifactPath.value
           val af = compileAnalysisFile.value
           IO.copyFile(original, artp)
-          if (af.exists) {
+          // skip zip manipulation if the artp is a blank file
+          if (af.exists && artp.length() > 0) {
             JarUtils.includeInJar(artp, Vector(af -> s"META-INF/inc_compile.zip"))
           }
           // val testStream = (test / streams).?.value

